crates/nu-cmd-lang/src/core_commands/echo.rs

```diff
@@ -43,6 +43,25 @@ little reason to use this over just writing the values as-is."#
         Ok(value.into_pipeline_data())
     }
 
+    fn run_const(
+        &self,
+        working_set: &StateWorkingSet,
+        call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let mut args = call.rest_const(working_set, 0)?;
+        let value = match args.len() {
```

Comment:
Thanks!  I think it's good to move duplicate code inside both `run` and `run_const` into a new function.
```rust
fn echo_impl(args: Vec<Value>) -> Result<PipelineData, ShellError> {
    let value = match args.len() {
            0 => Value::string("", call.head),
            1 => args.pop().expect("one element"),
            _ => Value::list(args, call.head),
        };
        Ok(value.into_pipeline_data())
}
```